apps/builder/app/builder/features/project-settings/meta-section.tsx

```diff
@@ -26,29 +28,41 @@ const imgStyle = css({
   borderColor: theme.colors.borderMain,
 });
 
-export const MetaSection = (props: {
-  meta: ProjectMeta;
-  onMetaChange: (value: ProjectMeta) => void;
-}) => {
-  const ids = useIds(["siteName", "favicon", "code"]);
-  const handleChange =
-    <Name extends keyof ProjectMeta>(name: Name) =>
-    (value: ProjectMeta[Name]) => {
-      props.onMetaChange({
-        ...props.meta,
-        [name]: value,
-      });
-    };
+const defaultMetaSettings: ProjectMeta = {
+  siteName: "",
+  faviconAssetId: "",
+  code: "",
+};
 
+export const MetaSection = () => {
+  const [meta, setMeta] = useState($pages.get()?.meta ?? defaultMetaSettings);
```

Comment:
please use function initializer, constant calling of .get in render deoptimizes stores

Reply:
do you mean using useStore?

Reply:
I mean useState(() => here)

Reply:
actually since I changed how we save, we do it now right away without waiting for blur or close, we can as well use useStore, because there is no need to have a stale/static state any more, wdyt?